servers/rendering/renderer_rd/renderer_canvas_render_rd.h

```diff
@@ -49,6 +49,9 @@ class RendererCanvasRenderRD : public RendererCanvasRender {
 		INSTANCE_DATA_UNIFORM_SET = 4,
 	};
 
+	static const int MAX_BATCH_TEXTURES = 31;
```

Comment:
We will need to set this dynamically based on `maxPerStageDescriptorSampledImages`. Unfortunately Vulkan only mandates that at least 16 be supported. GPUInfo seems to indicate that only supporting 16 is quite rare (https://vulkan.gpuinfo.org/displaydevicelimit.php?name=maxPerStageDescriptorSampledImages). Additionally, Apple A10 and earlier only support 31 total.

Based on the data in https://developer.apple.com/support/app-store/ we know that 9% of iPhone and 15% of iPads aren't on iOS 16 yet. While we can assume some of those are because of people failing to upgrade. I think we most are going to be using Apple A10(X) chips and lower. 

So while I'm not sure if it is worth supporting devices that can only have 16 textures. We need to at least have the ability to support devices that only support 31. The fallback is the same for both, so we should just implement it